src/core/filepath.h

```diff
@@ -130,11 +130,15 @@ class LIBFM_QT_API FilePath {
     }
 
     bool operator == (const FilePath& other) const {
-        if(gfile_ == other.gfile_) {
+        return operator==(other.gfile_.get());
+    }
+
+    bool operator == (GFile* other_gfile) const {
+        if(gfile_ == other_gfile) {
```

Comment:
I'm not sure about the compiler optimizations of such branching.... but logically this:
```c++
return gfile_ == other_file || (gfile_ && other_gfile && g_file_equal(gfile_.get(), other_gfile));
```
is the same as those 2 branches w/o the actual (explicit) branching.